oracle/oracle_test.go

```diff
@@ -93,6 +97,175 @@ func Test_Oracle_ManualSubscription(t *testing.T) {
 
 // TODO: Mix manual and automatic subscriptions
 
+// Simulates 100 slots with "AdvanceStateToNextSlot". Each slot is configured randomly with a
+// new sub, unsub or donation. The block proposed can be okproposal, missed or wrongfee.
+// these are all randomly set each block
+var numIterations = flag.Int("iterations", 100, "Number of iterations for the test")
+
+func Test_100_slots_test(t *testing.T) {
+	fmt.Printf("Number of iterations: %d\n", *numIterations)
+	//set new oracle instance
+	oracle := NewOracle(&config.Config{
+		Network:               "mainnet",
+		PoolAddress:           "0xdead000000000000000000000000000000000000",
+		UpdaterAddress:        "",
+		DeployedSlot:          uint64(50000),
+		CheckPointSizeInSlots: uint64(100),
+		PoolFeesPercent:       5,
+		PoolFeesAddress:       "0xfee0000000000000000000000000000000000000",
+		CollateralInWei:       big.NewInt(1000000),
+	})
+
+	subsIndex := make([]uint64, 0)
+	totalAssets := big.NewInt(0)
+	const seed int64 = 50000
+	rand.Seed(seed)
+	fmt.Printf("Execution SEED: %d\n", seed)
+	// main loop, iterates through 100 slots
+	for i := 0; i <= *numIterations; i++ {
+		newSubscription := make([]Subscription, 0)
+		newUnsubscription := make([]Unsubscription, 0)
+		don := make([]Donation, 0)
+		fmt.Println("NEW BLOCK:")
```

Comment:
can we use logrus for logging this? (see other code in the oracle)
* reading the logs doesn't seem very intuitive for me. can we have a single line per "event"? this would be easier to read
* also, if its empty, no need to print it. eg remove `Donation empty` etc.

feel free to use something like this.
```
log.WithFields(log.Fields{
		"ValidatorIndex (if any)":     xxx,
		"Reward (if applicable)": xx,
		"Slot":         xx,
	}).Info("Event: subscription/unsubs/block...")
```

Reply:
logrus implemented. Now logs look like this:

![image](https://user-images.githubusercontent.com/36164126/231834386-f12d8b27-4212-432d-9819-d0d1a88c1e11.png)

Each block logs is separated with an empty line. First log of each blocks is always "NEW BLOCK". The following log messages are the events (sub, unsub, don) to be mocked in this block. 

After the "Block type" log message, the oracle calls the "_AdvanceStateToNextSlot_" method and prints all the state changes.